### DISCUSSION TITLE
Toggle readonly mode in kie-editor

### DISCUSSION THREAD

**Discussion by sureshravi93** [apache/incubator-kie-issues#1611] [Q&A]
Hi
I am trying to build a vscode extension for dmn editor in similar to the incubator-kie-tools/dmn-vscode-extension. My objective is to first open the editor in readonly mode and switch to edit mode.  I have gone through the implementation of incubator-kie-tools/dmn-vscode-extension and incubator-kie-tools/vscode-extenstion and noticed that kogitoEditor_initRequest is used to initiate the editor and it can be called only once in the editor lifecycle. Is there a way that I could switch the editor to edit mode when the editor is already opened in read only mode.

**Comment by domhanak** [comment]
@sureshravi93 Hello, I think the only way is to reopen the editor with the property set, but I am not sure as I did not focus on this area at all in recent months.
I suggest to test the behavior and also consider looking at  https://github.com/apache/incubator-kie-tools/tree/main/packages/kie-editors-standalone

**Comment by tiagobento** [comment]
Please take a look at https://github.com/apache/incubator-kie-tools/tree/main/packages/dmn-editor-standalone too for the new DMN Editor.

Answering your question @sureshravi93: no, setting the read-only mode can only be done at instantiation at the moment.

If you wanna share more about your use case we might be able to better assist you. Thanks!
